ui/app/adapters/secret-engine.js

```diff
@@ -25,35 +25,61 @@ export default ApplicationAdapter.extend({
     return 'mounts';
   },
 
-  query(store, type, query) {
-    return this.ajax(this.internalURL(query.path), 'GET');
+  async query(store, type, query) {
+    let mountModel, configModel;
+    try {
+      mountModel = await this.ajax(this.internalURL(query.path), 'GET');
+      // if kv2 then add the config data to the mountModel
+      // version comes in as a string
+      if (mountModel.data.type === 'kv' && mountModel.data.options.version === '2') {
+        configModel = await this.ajax(this.urlForConfig(query.path), 'GET');
+        let data = mountModel.data;
+        data = { ...mountModel.data, ...configModel.data };
```

Comment:
Maybe we can simplify this directly? 
mountModel.data = { ...mountModel.data, ...configModel.data }

Reply:
Good call, thank you!